### PR TITLE
Resolving lint errors

### PR DESCRIPTION
**Pull request by ghazal-salehi**
`npm i eslint`
`npx eslint --fix src`
`npx eslint --fix gulp`
`npx eslint --fix gulpfile.babel.js`

All the errors are resolved by running above commands except bellow fixes were done manually:

- `src/assets/_project/_blocks/components/forms/qg-address-autocomplete.js`
- `gulp/publish-tasks/git.js`

### CODE REVIEW

File: src/assets/_project/_blocks/components/forms/qg-address-autocomplete.js [hunk 1]
```diff
@@ -36,13 +36,21 @@ export class QgAddressAutocomplete {
    **/
   _setValFromUrlParameters () {
     this.$form.find(':input:not(:checkbox):not(:radio), select, textarea').each(function () {
-      let name = $(this).attr('name');
+      const name = $(this).attr('name');
       let getParameterVal = qg.swe.getParameterByName($(this).attr('name'));
-      getParameterVal !== false ? $('[name="' + name + '"]').val(getParameterVal) : '';
+      if (getParameterVal !== false) {
```
Comment (ThrawnCA) [inline src/assets/_project/_blocks/components/forms/qg-address-autocomplete.js:41]: "if not...else" is a double negative, consider swapping it around.

File: src/assets/_project/_blocks/components/forms/qg-address-autocomplete.js [hunk 2]
```diff
@@ -36,13 +36,21 @@ export class QgAddressAutocomplete {
...
+        getParameterVal = $('[name="' + name + '"]').val(getParameterVal);
+      } else {
+        getParameterVal = '';
+      }
     }).end().find('input[type=checkbox], input[type=radio]').each(function () {
-      let name = $(this).attr('name');
+      const name = $(this).attr('name');
       let getParameterVal = qg.swe.getParameterByName(name);
-      getParameterVal !== false ? $('[value="' + getParameterVal + '"]').prop('checked', true) : '';
+      if (getParameterVal !== false) {
```
Comment (ThrawnCA) [inline src/assets/_project/_blocks/components/forms/qg-address-autocomplete.js:49]: "if not...else" is a double negative, consider swapping it around.

File: src/assets/_project/lib/ext/butterfly/jquery.butterfly.js
Comment (ThrawnCA) [inline src/assets/_project/lib/ext/butterfly/jquery.butterfly.js:1]: I don't think we should be linting library modules.

File: src/docs/assets/js/prism.js
Comment (ThrawnCA) [inline src/docs/assets/js/prism.js:1]: Don't lint minified files. Adding spaces defeats the purpose of minification.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

File: src/assets/_project/lib/ext/butterfly/jquery.butterfly.js
Comment (ghazal-salehi) [reply src/assets/_project/lib/ext/butterfly/jquery.butterfly.js:1]: When I tried running `npm run publish-test-npm` locally, lint errors stopped building remotely. `npm run build` finishes locally though (still showing the errors). Can you try `npm run publish-test-npm` locally on develop branch to see if that works for you? If it does we can leave some of the lint errors like the ones in the minified js and external libraries.

I checked older versions of eslint package as well. Still shows 700 errors on dev branch. Not sure why all of a sudden these started happening.

File: src/docs/assets/js/prism.js
Comment (ghazal-salehi) [reply src/docs/assets/js/prism.js:1]: There are more ext libraries. I will revert them.

File: src/assets/_project/_blocks/components/accordion/qg-accordion.js [hunk 3]
```diff
@@ -162,13 +162,13 @@ export class QgAccordion {
    * @return {undefined}
    **/
   legacyAccordion(){
-    let self = this;
-    let accItem = $('.qg-accordion:not(.qg-accordion-v2)').find('article');
+    const self = this;
```
Comment (duttonw) [inline src/assets/_project/_blocks/components/accordion/qg-accordion.js:165]: don't think these can be const. Is this code stood up multiple times for each accordion?

File: src/assets/_project/_blocks/components/carousel/qg-carousel.js [hunk 4]
```diff
@@ -2,16 +2,16 @@
 'use strict';
 
 (function ($) {
-  let carousels = [];
+  const carousels = [];
```
Comment (duttonw) [inline src/assets/_project/_blocks/components/carousel/qg-carousel.js:5]: can't be const, we need to be adding items to the array.

**Review by duttonw — CHANGES_REQUESTED** [review]
Const is only used if it will never change.

There is too many places where  ```let``` is the only choice.  (unless this javascript function code is creation immutable objects instead of being 'function' calls.

https://www.freecodecamp.org/news/differences-between-var-let-const-javascript/#:~:text=If%20you%20never%20want%20a,is%20the%20keyword%20for%20you

File: src/docs/assets/js/prism.js
Comment (duttonw) [reply src/docs/assets/js/prism.js:1]: ext libraries should be ignored from lint checkers. same as the dist folder etc.

**Comment by ghazal-salehi** [comment]
Should now be good to review again. Sorry about too many commits!

**Comment by ghazal-salehi** [comment]
> Const is only used if it will never change.
> 
> There is too many places where `let` is the only choice. (unless this javascript function code is creation immutable objects instead of being 'function' calls.
> 
> https://www.freecodecamp.org/news/differences-between-var-let-const-javascript/#:~:text=If%20you%20never%20want%20a,is%20the%20keyword%20for%20you

I updated some references in accordion.js and carousel.js to use `var` instead of `let` to pass lint check.
npx eslint --fix seems to be not so reliable after all.

> Const is only used if it will never change.
> 
> There is too many places where `let` is the only choice. (unless this javascript function code is creation immutable objects instead of being 'function' calls.
> 
> https://www.freecodecamp.org/news/differences-between-var-let-const-javascript/#:~:text=If%20you%20never%20want%20a,is%20the%20keyword%20for%20you

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**